slsa/v1.0/L1/buildtype.md

```diff
@@ -0,0 +1,115 @@
+<!--
+SPDX-FileCopyrightText: 2022-2024 TII (SSRC) and the Ghaf contributors
+SPDX-License-Identifier: CC-BY-SA-4.0
+-->
+
+# Build Type: Jenkins
+
+This is a [SLSA Provenance](https://slsa.dev/provenance/v1) `buildType` that describes the execution of a Jenkins workflow.
+
+This build type was inspired by the Cimon Jenkins build type described [here](https://docs.cimon.build/provenance/buildtypes/jenkins/v1).
+
+## Description
+
+```json
+"buildType": "https://github.com/tiiuae/ghaf-infra/blob/main/slsa/v1.0/L1/buildtype.md"
+```
+
+The Jenkins build type describes the execution of a Jenkins workflow that builds a software artifact.
+
+## Build Definition
+
+### External parameters
+
+All external parameters are REQUIRED unless empty.
+
+Parameter             | Type   | Description
+--------------------- | ------ | -----------
+target                | object | The target that was built
+target.name           | string | The full name of the nix target
+target.repository     | string | URI of the git repository (if exists)
+target.ref            | string | A git reference to the commit (if exists)
+workflow              | object | The workflow that was run
+workflow.name         | string | The full name of the Jenkins job
+workflow.repository   | string | URI of the git repository (if exists)
+workflow.ref          | string | A git reference to the commit (if exists)
+workflow.filePath     | string | -
+job                   | object | The currently running Jenkins job
+buildRun              | object | The specific build that generated the provenance
+
+Example:
+
+```json
+"externalParameters": {
+    "target": {
+        "name": "#packages.x86_64-linux.generic-x86_64-debug",
+        "repository": "https://github.com/tiiuae/ghaf.git",
+        "ref": "main"
```

Comment:
"ref" should never be a floating one like branch HEAD, so should not use such in example either. Could be something that's clearly a tag.

Reply:
Ideally this would be a commit hash but currently our system only gives us the branch

Reply:
So we don't know the exact commit / sources that the build has been made from? Has SLSA anything to say about that? @alextserepov

Reply:
I have made changes to the pipeline that allow using the commit hash here